### PR TITLE
ci: pick up cilium-cli v0.12.0 for master, v1.11 and v1.12 workflows

### PR DESCRIPTION
**Pull request by tklauser**
Release notes: https://github.com/cilium/cilium-cli/releases/tag/v0.12.0

Note for backporters: only the following files exist in `v1.11` and `v1.12` branches for which need to be backported:

```
.github/workflows/conformance-ingress.yaml
.github/workflows/conformance-kind.yaml
.github/workflows/tests-smoke-ipv6.yaml
.github/workflows/tests-smoke.yaml
```

### CODE REVIEW

**Review by sayboras — APPROVED** [review]
LGTM ✅ (I didn't check the temp commit)

**Comment by tklauser** [comment]
All conformance tests passed on `master`, removing temporary test commit.

**Comment by joestringer** [comment]
I guess that when you tested this, you were effectively testing the master workflow file changes but not the v1.11 or v1.12 ones? Do you have any reason to believe there could be any impact on the older branches?

**Review by joestringer — APPROVED** [review]
Changes LGTM, assuming that CLI v0.12.0 is compatible with both Cilium v1.11.x and v1.12.y. Per my comment above, I suspect that those two have not yet been tested with this PR so you may want to either do additional testing somehow, or be around when it merges so you can double-check that it's working (+revert if it doesn't do the right thing).

**Comment by tklauser** [comment]
> Changes LGTM, assuming that CLI v0.12.0 is compatible with both Cilium v1.11.x and v1.12.y. Per my comment above, I suspect that those two have not yet been tested with this PR so you may want to either do additional testing somehow, or be around when it merges so you can double-check that it's working (+revert if it doesn't do the right thing).

Thanks for the review. As you noted, this was only tested on `master`. But I suspect this should work for v1.11 and v1.12 since it was tested with both releases in the cilium-cli repo's CI before release, so I don't think additional testing should be needed before merge. In any case I'll be around to send a revert in the unlikely case this would break v1.11/v1.12 workflows.

**Comment by nbusseneau** [comment]
@tklauser This PR depends on changes from #20203, which was not backported to `v1.11`. I will backport the dependent PR as well.

**Pull request merged**
